### PR TITLE
chore(weave): Add join and group by to orm

### PR DESCRIPTION
**Pull request by jwlee64**
Add join and group by to orm

### CODE REVIEW

**Review by tssweeney — APPROVED** [review]

File: weave/trace_server/tests/test_orm.py [hunk 1]
```diff
@@ -173,3 +174,117 @@ def test_select_fields():
...
+
+    assert (
+        prepared.sql
+        == """SELECT id, name
+FROM users
+JOIN roles ON (table1.id = table2.id)"""
+    )
+
+
+def test_special_join():
```
Comment (jamie-rasmussen) [inline weave/trace_server/tests/test_orm.py:225]: Maybe a more descriptive name, `test_join_with_join_type`

File: weave/trace_server/tests/test_orm.py [hunk 2]
```diff
@@ -173,3 +174,117 @@ def test_select_fields():
...
+        ],
+    )
+
+    select = (
+        table1.select()
+        .fields(["id", "name"])
+        .join(
+            "",
+            table2,
+            tsi.Query(
```
Comment (jamie-rasmussen) [inline weave/trace_server/tests/test_orm.py:202]: I know we want to guard against SQL injection, but it feels wordy and weird to me to use this MongoDB syntax for the join condition.

Comment (jwlee64) [reply weave/trace_server/tests/test_orm.py:202]: I too think its wordy, but i feel like having to parse which conditions require the mongo syntax and which dont feels annoying as well, vs everything in the mongo syntax

File: weave/trace_server/orm.py [hunk 3]
```diff
@@ -210,18 +223,27 @@ class Select:
...
         self._query = None
         self._order_by = None
         self._limit = None
         self._offset = None
+        self._group_by = None
+
+    def join(self, join_type: str, table: Table, query: tsi.Query) -> "Select":
+        self.joins.append(Join(join_type, table, query))
+        for col in table.cols:
+            self.all_columns.append(col.dbname())
```
[the remark this answers is not in the document]
Comment (jwlee64) [reply weave/trace_server/orm.py:245]: yes you do I made a change in the orm file that allows for table prefixes

**Pull request merged**